File: JsonSchemaAndStructs/KiteConnect-Kotlin-Structs/order_margins.kt
// To parse the JSON, install kotlin's serialization plugin and do:
//
// val json         = Json(JsonConfiguration.Stable)
// val orderMargins = json.parse(OrderMargins.serializer(), jsonString)

package OrderMargins

import kotlinx.serialization.*
import kotlinx.serialization.json.*
import kotlinx.serialization.descriptors.*
import kotlinx.serialization.encoding.*

@Serializable
data class OrderMargins (
    val data: List<Datum>? = null,
    val status: String? = null
)

@Serializable
data class Datum (
    val additional: Long? = null,
    val bo: Long? = null,
    val cash: Long? = null,
    val exchange: String? = null,
    val exposure: Long? = null,

    @SerialName("option_premium")
    val optionPremium: Long? = null,

    val pnl: Pnl? = null,
    val span: Long? = null,
    val total: Double? = null,
    val tradingsymbol: String? = null,
    val type: String? = null,

    @SerialName("var")
    val datumVar: Double? = null
)

@Serializable
data class Pnl (
    val realised: Long? = null,
    val unrealised: Long? = null
)

File: JsonSchemaAndStructs/KiteConnect-Rust-Structs/order_margins.rs
// Example code that deserializes and serializes the model.
// extern crate serde;
// #[macro_use]
// extern crate serde_derive;
// extern crate serde_json;
//
// use generated_module::[object Object];
//
// fn main() {
//     let json = r#"{"answer": 42}"#;
//     let model: [object Object] = serde_json::from_str(&json).unwrap();
// }

extern crate serde_derive;

#[derive(Debug, Serialize, Deserialize)]
pub struct OrderMargins {
    pub data: Option<Vec<Datum>>,
    pub status: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Datum {
    pub additional: Option<i64>,
    pub bo: Option<i64>,
    pub cash: Option<i64>,
    pub exchange: Option<String>,
    pub exposure: Option<i64>,
    pub option_premium: Option<i64>,
    pub pnl: Option<Pnl>,
    pub span: Option<i64>,
    pub total: Option<f64>,
    pub tradingsymbol: Option<String>,
    #[serde(rename = "type")]
    pub datum_type: Option<String>,
    pub var: Option<f64>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Pnl {
    pub realised: Option<i64>,
    pub unrealised: Option<i64>,
}
